src/python/pants/backend/go/util_rules/assembly.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import PurePath
+
+from pants.backend.go.util_rules.sdk import GoSdkProcess
+from pants.engine.fs import CreateDigest, Digest, FileContent, MergeDigests
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+
+
+@dataclass(frozen=True)
+class AssemblyPreCompilation:
+    merged_compilation_input_digest: Digest
+    assembly_digests: tuple[Digest, ...]
+    EXTRA_COMPILATION_ARGS = ("-symabis", "./gensymabis")
```

Comment:
A minor fix that I didn't make when I was working on this stuff: The file should be called `symabis` and not `gensymabis` since `gensymabis` is the name of the command that generates the file.